pkg/3scale/amp/operator/zync_reconciler.go

```diff
@@ -21,13 +22,18 @@ func NewZyncReconciler(baseAPIManagerLogicReconciler *BaseAPIManagerLogicReconci
 }
 
 func (r *ZyncReconciler) Reconcile() (reconcile.Result, error) {
+	ampImages, err := AmpImages(r.apiManager)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
 	zync, err := Zync(r.apiManager, r.Client())
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
 	// Zync Que Role
-	err = r.ReconcileRole(zync.QueRole(), reconcilers.CreateOnlyMutator)
+	err = r.ReconcileRole(zync.QueRole(), reconcilers.RoleRuleMutator)
```

Comment:
You're correct that we used to only create the Role and nothing more. However that Role's rule used to be for [DeploymentConfigs](https://github.com/3scale/3scale-operator/blame/de98237e57d7a68a4f44c5a27edabeebd8184136/pkg/3scale/amp/component/zync.go#L112) but it now needs to apply for [Deployments](https://github.com/3scale/3scale-operator/blob/845cddc4d9cc1405c6f10b65f50c727063e438d5/pkg/3scale/amp/component/zync.go#L113).

I thought it would be better to update the Role rather than delete it and create a new one which is why we need the new mutator to update the RBAC.

Reply:
thanks